File: src/main/java/com/dyrosoft/kvbparser/merger/StationWithLineMergerFunc.java
package com.dyrosoft.kvbparser.merger;

import com.dyrosoft.kvbparser.models.Line;
import com.dyrosoft.kvbparser.models.Station;
import com.dyrosoft.kvbparser.models.StationDetails;

import java.util.List;

import rx.functions.Func2;

public class StationWithLineMergerFunc
        implements Func2<List<Line>, Station, StationDetails> {

    @Override
    public StationDetails call(final List<Line> lines, final Station station) {
        return new StationDetails(station.getId(), station.getName(), lines);
    }
}

File: src/main/java/com/dyrosoft/kvbparser/merger/StationWithLineMergerFunc.kt
package com.dyrosoft.kvbparser.merger

import com.dyrosoft.kvbparser.models.Line
import com.dyrosoft.kvbparser.models.Station
import com.dyrosoft.kvbparser.models.StationDetails

import rx.functions.Func2

internal class StationWithLineMergerFunc : Func2<List<Line>, Station, StationDetails> {

    override fun call(lines: List<Line>, station: Station) = StationDetails(station.id, station.name, lines)
}
